src/MizarData/include/MizarData/BaselineOrComparison.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+#define MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+
+#include <type_traits>
+#include <utility>
+
+namespace orbit_mizar_data {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+template <typename T>
+class BaselineOrComparison {
```

Comment:
This class needs a comment.

Reply:
Actually I don't really understand why you need all these classes. I sense you want to have stronger typing but all these templates are just telling me it's not really worth it.

Reply:
Thanks for the question. I've been reluctant to introduce these. But then [this](https://github.com/google/orbit/pull/3779#discussion_r895484726) happened. That's it, in numerous places we will have pairs of instances of the same type, one related to the baseline, the other -- to comparison. And if I have erroneously swapped them at such an early stage of development, this will become a major pain down the road. These classes catch this kind of bugs in compile-time. 

But everything comes at a price. Like this:
```
template <template <typename> typename Wrapper>
  [[nodiscard]] Wrapper<orbit_client_data::ScopeStats> MakeFrameTrackStats(
      const Wrapper<PairedData>& data,
      const Wrapper<HalfOfSamplingWithFrameTrackReportConfig>& config)
```
here we have to (verbosely) enforce that we wish to have objects of some types wrapped in the same wrapper. And this does "help the code". 

OTOH, there are getters in `SamplingWithFrameTrackComparisonReport` and I do admit they're more for the sake of an exercise than for anything else. I will probably replace them with normal getters before merging -- I just want somebody to have a look and, perhaps, give an insightful tip.

Reply:
I am here with @dpallotti. Readability is IMO drastically reduced. I'd rather introduce proper wrapper classes for your "pairs". That way you reduce the risk of mixing up comparison and baseline values of the "pairs" to their constructor calls. Still not great, but less likely to break and still pretty readably without all the template magic.

Reply:
Since this is more and more evolving we could also consider replacing `Baseline` and `Comparison` by a proper strong-typedef-like container that lives in OrbitBase, is very generic, and keeps the complexity out of the domain specific code. I'm think about something like this:

```cpp
template<typename Tag, typename T>
struct Typedef {
  // constructors, operators, etc.
private:
  T value_;
};
```

Baseline would then be defined like this:

```cpp
struct BaselineTag{};

template<typename T>
using Baseline = orbit_base<BaselineTag, T>;
```

This would mean though that there is no support for a hierarchy, so no `BaselineOrComparison`. So functions currently taking `BaselineOrComparison<T>` would have to take a `T` instead.

Reply:
@beckerhe I like the idea. Two questions. 
1. Should I move this to OrbitBase? Do you wish to make these typedefs universally accessible in Orbit? 
2. Why should we prefer a two-argument type over inheritance? We will not need template template argument, is that the point? But it will add to the size of the structure. It's +4 bytes with clang. Eventually we will wish to wrap ints into this (e.g. some ids). And the overhead will be tangible.

Reply:
... and again, it will be in a separate PR.

Reply:
Yes. I think this should be in OrbitBase.

I don't really understand your second question. Are you talking about the Tag? It's needed to generate distinct types to make it possible to declare a string Typedef with a simple using declaration as demonstrated above.

Reply:
I see how you suggest to do this. I am not sure what are the pros compared to inheritance.